### PR TITLE
feat(images): update lightning nodes, add cli completion

### LINKED ISSUE #375: Feature Request: CLI autocomplete for lightning & bitcoin nodes
**Is your feature request related to a problem? Please describe.**
I like to use CLI when interacting with nodes. Sadly autocompletion doesn't work. 

**Describe the solution you'd like**
When pressing tab button bitcoin-cli or lncli should show all available options

**Additional context**
I can try implementing this if you give me some pointers

### PR DESCRIPTION
**Pull request by jonafll**
Closes #375, #540 

### Description

This PR updates the images of lnd, c-lightning and eclair to the latest version. It also adds support for cli completion in the terminal. Of course, before this PR can be merged, the new images must be available on Docker Hub.

LND v0.14.1-beta => v0.14.2-beta
C-Lightning v0.10.0 => v0.10.2
Eclair v0.6.1 => v0.7.0

### CODE REVIEW

File: docker/clightning/Dockerfile [hunk 1]
```diff
@@ -99,19 +99,29 @@ RUN git clone --recursive --branch=v${CLN_VERSION} https://github.com/ElementsPr
 # Original line: ARG DEVELOPER=0
 ARG DEVELOPER=1
 ####################
+
+ENV PYTHON_VERSION=3
+
+#################### Polar Modification
+# Use fixed version of mistune, because the latest version breaks the mrkd package
+# https://github.com/ElementsProject/lightning/issues/4973
+RUN pip3 install mrkd mistune==0.8.4
```
Comment (jamaljsr) [inline docker/clightning/Dockerfile:108]: There's been a more recent update in the `lightning` repo to fix the `mrkd` package 
```
RUN apt-get install -y --no-install-recommends python3-dev
RUN pip3 install -U pip && pip3 install -r requirements.lock
```

See https://github.com/ElementsProject/lightning/pull/4969

File: docker/nodes.json [hunk 2]
```diff
@@ -2,8 +2,9 @@
   "version": 34,
```
Comment (jamaljsr) [inline docker/nodes.json:2]: This version should be incremented when updating the node versions.

File: src/utils/constants.ts [hunk 3]
```diff
@@ -219,8 +219,9 @@ export const defaultRepoState: DockerRepoState = {
   version: 34,
```
Comment (jamaljsr) [inline src/utils/constants.ts:219]: This version should be incremented when updating the node versions.

File: docker/lnd/Dockerfile [hunk 4]
```diff
@@ -12,7 +12,17 @@ RUN curl -SLO https://github.com/lightningnetwork/lnd/releases/download/v${LND_V
   && tar -xzf *.tar.gz -C /opt \
   && rm *.tar.gz
 
+RUN curl -SLO https://github.com/lightningnetwork/lnd/releases/download/v${LND_VERSION}/lnd-linux-amd64-v${LND_VERSION}.tar.gz \
```
Comment (jamaljsr) [inline docker/lnd/Dockerfile:15]: Do we need to download  and extract the LND tar file again?

File: docker/eclair/Dockerfile [hunk 5]
```diff
@@ -98,7 +96,14 @@ ENV ECLAIR_DATADIR=/home/eclair/
...
+RUN curl -SLO https://raw.githubusercontent.com/ACINQ/eclair/master/contrib/eclair-cli.bash-completion \
+  && mkdir /etc/bash_completion.d \
+  && mv eclair-cli.bash-completion /etc/bash_completion.d/ \
+  && curl -SLO https://raw.githubusercontent.com/scop/bash-completion/master/bash_completion \
+  && mv bash_completion /usr/share/bash-completion/
+
 COPY docker-entrypoint.sh /entrypoint.sh
+COPY bashrc /home/eclair/.bashrc
 
 RUN chmod a+x /entrypoint.sh
```
Comment (jamaljsr) [inline docker/eclair/Dockerfile:108]: There is an issue with Eclair nodes. After opening a channel and mining blocks, the channel state remains at `WAIT_FOR_FUNDING_CONFIRMED` for a long time. This prevents payments from being routed over the new channel immediately. As of Eclair v0.6.2, we should use the `zmqpubhashblock` bitcoind config option. Check out the [release notes](https://github.com/ACINQ/eclair/blob/master/docs/release-notes/eclair-v0.6.2.md). 

I was able to resolve the issue in my branch by updating two lines in the [constants.ts](https://github.com/jamaljsr/polar/commit/2128993d6558c0c930ceec28ecb98d4b96d3a8c7#diff-e8d0a358ae2fb090c56f00a058050983cc9691b7d3622ddd443170017123e7d9L157) file. Can you bring these changes over to this PR?

**Review by jamaljsr — CHANGES_REQUESTED** [review]
Thank you so much for making these updates. Funny thing is I was actually working on these same changes myself today. Since this PR also contains the bash completions, I'd prefer to move forward with yours instead of mine.

I've pushed new images to Docker Hub based on the Dockerfiles in this PR.

Can you split up the commit into three separate commits? One for each node update. This just makes it easier to revert if needed in the future.

File: docker/lnd/Dockerfile (hunk 4 above)
Comment (jonafll) [reply docker/lnd/Dockerfile]: copy & paste ;)

**Comment by jamaljsr** [comment]
Thanks for resolving all of those issues. I have tested again and the new images work great.

I do have one request before merging however. It would be better to drop the commits https://github.com/jamaljsr/polar/pull/541/commits/e2cebf5d7019cb79342266272ad8937b1526985a and https://github.com/jamaljsr/polar/pull/541/commits/b1e71e272d0fec1a16fd038358b7b6f351b77897 from the git history. You can use `git rebase -i master` to drop them, then force push to your branch on github.

**Review by jamaljsr — APPROVED** [review]
Thanks for making these updates. 🥇

**Pull request merged**